tests/unittests/test_handler/test_handler_write_files.py

```diff
@@ -36,13 +40,85 @@
     '/tmp/message': "hi mom line 1\nhi mom line 2\n",
 }
 
+VALID_SCHEMA = {
+    'write_files': [
+        {'append': False, 'content': 'a', 'encoding': 'gzip', 'owner': 'jeff',
+         'path': '/some', 'permissions': '0777'}
+    ]
+}
+
+
+@skipUnlessJsonSchema()
+@mock.patch('cloudinit.config.cc_write_files.write_files')
+class TestWriteFilesSchema(CiTestCase):
+
```

Comment:
Should we pytest paramtrize this?

Reply:
Probably, but I would like to defer that since we are also searching logs in each of these tests. There are pytest logging fixture incompatibilities that we have had to fix on Xenial and Bionic in ubuntu-advantage-client and I think we need to introduce a common logging fixture here for cloud-init first to avoid failures in CI on older releases.

Reply:
> There are pytest logging fixture incompatibilities that we have had to fix on Xenial and Bionic in ubuntu-advantage-client

Yep, the `caplog` fixture is not available in the version of pytest in xenial. However, unlike ubuntu-advantage-client, we don't have to support trusty, which means we could consider using https://launchpad.net/ubuntu/+source/pytest-catchlog which is present in xenial. Per [its GitHub page](https://github.com/eisensheng/pytest-catchlog), it was merged into pytest core in 3.3 (the version in bionic).

(I don't hold an opinion about what we should do here, just saw this come through my inbox and figured I'd fill in some background.)

Reply:
Thanks @OddBloke I'll look at using pytest-catchlog in a separate PR, then can work toward parameterizing some of these log-related unittests with pytest.